src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -315,6 +349,54 @@ public static void ExecutingAction(this ILogger logger, ActionDescriptor action)
             _actionExecuting(logger, action.DisplayName, null);
         }
 
+        public static void AuthorizationFiltersExecutionPlan(this ILogger logger, IEnumerable<IFilterMetadata> filters)
+        {
+            var authorizationFilters = filters.Where(f => f is IAuthorizationFilter || f is IAsyncAuthorizationFilter);
+            LogFilterExecutionPlan(logger, "authorization", authorizationFilters);
+        }
+
+        public static void ResourceFiltersExecutionPlan(this ILogger logger, IEnumerable<IFilterMetadata> filters)
+        {
+            var resourceFilters = filters.Where(f => f is IResourceFilter || f is IAsyncResourceFilter);
+            LogFilterExecutionPlan(logger, "resource", resourceFilters);
+        }
+
+        public static void ActionFiltersExecutionPlan(this ILogger logger, IEnumerable<IFilterMetadata> filters)
+        {
+            var actionFilters = filters.Where(f => f is IActionFilter || f is IAsyncActionFilter);
+            LogFilterExecutionPlan(logger, "action", actionFilters);
+        }
+
+        public static void ExceptionFiltersExecutionPlan(this ILogger logger, IEnumerable<IFilterMetadata> filters)
+        {
+            var exceptionFilters = filters.Where(f => f is IExceptionFilter || f is IAsyncExceptionFilter);
+            LogFilterExecutionPlan(logger, "exception", exceptionFilters);
+        }
+
+        public static void ResultFiltersExecutionPlan(this ILogger logger, IEnumerable<IFilterMetadata> filters)
+        {
+            var resultFilters = filters.Where(f => f is IResultFilter || f is IAsyncResultFilter);
+            LogFilterExecutionPlan(logger, "result", resultFilters);
+        }
+
+        public static void BeforeExecutingMethodOnFilter(
+            this ILogger logger,
+            string filterType,
+            string methodName,
+            IFilterMetadata filter)
+        {
+            _beforeExecutingMethodOnFilter(logger, filterType, methodName, filter.GetType().ToString(), null);
```

Comment:
I realized that I should avoid doing the '.ToString()' here as it does allocate before even the log level is checked. I could just pass in the `Type` here and the '[log values formatter](https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs#L140)' would take care of converting it to string...I am making changes to both of my logging PRs